From 90dadfe8b7b666967863802793418bcffc1c8874 Mon Sep 17 00:00:00 2001
From: Reino17 <rwijnsma@xs4all.nl>
Date: Sun, 30 Aug 2020 18:48:46 +0200
Subject: [PATCH 2/4] avformat/udp: revert 53aa766 for winxp compatibility

Otherwise you'd get "The procedure entry point CancelIoEx could not be located in the dynamic link library KERNEL32.dll" while running ffmpeg.exe, ffplay.exe, or ffprobe.exe, because 'CancelIoEx()' is only available on Windows Vista and later. See https://github.com/FFmpeg/FFmpeg/commit/53aa76686e7ff4f1f6625502503d7923cec8c10e and https://trac.ffmpeg.org/ticket/5717. This obviously doesn't fix the ticket, but simply reverts the commit.
---
 libavformat/udp.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/libavformat/udp.c b/libavformat/udp.c
index 30d8041..1f6681b 100644
--- a/libavformat/udp.c
+++ b/libavformat/udp.c
@@ -1074,17 +1074,8 @@ static int udp_close(URLContext *h)
     if (s->thread_started) {
         int ret;
         // Cancel only read, as write has been signaled as success to the user
-        if (h->flags & AVIO_FLAG_READ) {
-#ifdef _WIN32
-            /* recvfrom() is not a cancellation point for win32, so we shutdown
-             * the socket and abort pending IO, subsequent recvfrom() calls
-             * will fail with WSAESHUTDOWN causing the thread to exit. */
-            shutdown(s->udp_fd, SD_RECEIVE);
-            CancelIoEx((HANDLE)(SOCKET)s->udp_fd, NULL);
-#else
+        if (h->flags & AVIO_FLAG_READ)
             pthread_cancel(s->circular_buffer_thread);
-#endif
-        }
         ret = pthread_join(s->circular_buffer_thread, NULL);
         if (ret != 0)
             av_log(h, AV_LOG_ERROR, "pthread_join(): %s\n", strerror(ret));
-- 
2.8.3

